### ISSUE TITLE
Inconsistency in Accessing results.csv Path in engine/trainer.py since v8.0.174

### ISSUE THREAD

**Issue by SilverStarCoder** [ultralytics/ultralytics#4901]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Since **v8.0.174**, I've noticed a direct path reference to `results.csv` in the `save_model()` function of [engine/trainer.py](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/engine/trainer.py):
```
import pandas as pd
metrics = {**self.metrics, **{'fitness': self.fitness}}
results = {k.strip(): v for k, v in pd.read_csv(self.save_dir / 'results.csv').to_dict(orient='list').items()}
```
However, the BaseTrainer class in the same file has an attribute named `csv` which I believe is intended to provide flexibility for specifying custom paths to the results CSV file:
`self.csv = self.save_dir / 'results.csv'`
Using the csv attribute in the `save_model() `function, as shown below, would be more consistent and would respect custom path settings:
`results = {k.strip(): v for k, v in pd.read_csv(self.csv).to_dict(orient='list').items()}`
By ensuring the `results.csv` path is managed using the `csv `attribute, it would potentially offer a solution for users encountering path-related issues, especially when a custom path is desired for unique setups. For example, in my case, because of [this known issue on Databricks](https://kb.databricks.com/dbfs/errno95-operation-not-supported)., I specified a temporary file path for the `results.csv` file and then moved it after training was completed. Now, because of the latest update, I am unable to do this now, as I get a `FileNotFoundError` as it now always tries to read the `self.save_dir / 'results.csv'` , which  doesn't exist in my case. 

### Environment

- YOLO: Ultralytics YOLO v8.0.175🚀
- Azure Databricks

### Minimal Reproducible Example

N/A

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@SilverStarCoder great point, thanks for bringing this up! I'd completely forgotten about the results.csv attribute, I'll make this change as you recommended.

**Comment by glenn-jocher** [comment]
@SilverStarCoder great news 😃! Your original issue may now be resolved ✅ in PR #4894. I've made this fix in https://github.com/ultralytics/ultralytics/pull/4894/commits/b675b96f4a34057d6743641221e2185505c3d083 which will be part of the 8.0.179 release later today.  

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
